internal/git/gitsync_processor.go

```diff
@@ -127,14 +132,24 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1alpha1.GitSync
 		err = tree.Files().ForEach(func(f *object.File) error {
 			logger.Debugw("read file", "file_name", f.Name)
 			// TODO: this currently assumes that one file contains just one manifest - modify for multiple
-			manifest, err := f.Contents()
-			if err != nil {
-				logger.Errorw("cannot get file content", "filename", f.Name, "err", err)
-				return err
-			}
+			// TODO :need to address the valid file extension issue
+			if validations.IsValidManiFestFile(f.Name) {
```

Comment:
```suggestion
			if validations.IsValidManifestFile(f.Name) {
```

Reply:
Do we only call this for the initial clone? Will we take care of new files that get introduced in subsequent commits?

Reply:
Yes its been added here -https://github.com/numaproj-labs/numaplane/blob/6e8fedead839d071708ef6e3d36281d58a017440/internal/git/gitsync_processor.go#L242

Reply:
I'm referring to `IsValidManifestFile()` function - it seems like that's only called during the initial clone, unless I'm mistaken.